enterprise/views.py

```diff
@@ -127,27 +185,52 @@ def get(self, request):
 
         data = {
             'platform_name': platform_name,
-            'sso_provider': customer.name,
+            'ec_name': customer.name,
             'data_sharing_consent': 'required' if required else 'optional',
             "messages": {
                 "warning": self.get_warning(customer.name, platform_name, required),
                 "note": self.get_note(customer.name, required),
             },
+            "course_id": None
         }
         return render_to_response('grant_data_sharing_permissions.html', data, request=request)
 
-    def post(self, request):
+    def get(self, request):
         """
-        Process the above form.
+        Render a form to collect user input about data sharing consent.
         """
         # Verify that all necessary resources are present
         verify_edx_resources()
+        course = request.GET.get('course_id')
+        if course:
+            return self.get_course_specific_consent(request, course)
+        return self.get_account_consent(request)
+
+    def post_course_specific_consent(self, request, course_id, consent_provided):
+        """
+        Save user-provided consent directives for a particular course enrollment.
+        """
+        if not request.user.is_authenticated():
```

Comment:
It's not entirely clear to me when authentication is required and when it is not -- I can see we are performing a check here but not in the other `post` operations.